modules/ingest-common/src/test/java/org/elasticsearch/ingest/common/ScriptProcessorTests.java

```diff
@@ -58,7 +59,7 @@ public void testScripting() throws Exception {
         doAnswer(invocationOnMock ->  {
             ingestDocument.setFieldValue("bytes_total", randomBytesTotal);
```

Comment:
one thing we do not check here is that the Map passed in as the ctx is actually 

```
IngestDocument#getSourceAndMetadata()
```
Do you think that would be useful here?

We do have checks in `rest-api-spec/test/ingest/50_script_processor_using_painless.yml` that we are actually modifying the document, so that may be sufficient. Seems like it would be nice to validate this from unit tests too

Reply:
@talevy makes sense + is trivial to do => sec adjusting :)

Reply:
@talevy done in 89874c9 :)

Reply:
cooool. thanks!